File: src/test/java/com/larryhsiao/clotho/dgist/HexStringTest.java
package com.larryhsiao.clotho.dgist;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.nio.charset.StandardCharsets;

/**
 * Test for hex given string
 */
public class HexStringTest {
    /**
     * Normal case
     */
    @Test
    void normal() {
        Assertions.assertEquals(
            "414243",
            new HexString("ABC".getBytes(StandardCharsets.UTF_8)).value()
        );
    }
}

File: src/test/java/com/larryhsiao/clotho/encryption/HexStringTest.kt
package com.larryhsiao.clotho.encryption

import com.larryhsiao.clotho.dgist.HexString
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

/**
 * Test for [HexString].
 */
class HexStringTest {
    /**
     * Check the hex result.
     */
    @Test
    fun simple() {
        assertEquals(
            "414243",
            HexString("ABC".toByteArray()).value()
        )
    }
}